src/components/Table/ContextMenu/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import _find from "lodash/find";
+import { PopoverProps } from "@mui/material";
+
+import { getFieldProp } from "@src/components/fields";
+import { useProjectContext } from "@src/contexts/ProjectContext";
+
+import { MenuContents } from "./MenuContent";
+
+export type SelectedCell = {
+  rowIndex: number;
+  colIndex: number;
+};
+
+export type ContextMenuRef = {
+  selectedCell: SelectedCell;
+  setSelectedCell: React.Dispatch<React.SetStateAction<SelectedCell | null>>;
+  anchorEl: HTMLElement | null;
+  setAnchorEl: React.Dispatch<
+    React.SetStateAction<PopoverProps["anchorEl"] | null>
+  >;
+};
```

Comment:
Side note: this state ref via context hack is a great candidate for migration to Jotai in the future